File: Swift-C++-Testsuite/test/structsAndClasses/Swift/Structs/StructSelfDisambiguity.swift
//
//  File.swift
//
//
//  Created by Jan-Niclas Strüwer on 26.10.22.
//

import Foundation

@main
struct MyMain {

    static func main() {
        let somePoint = Point(x: 4.0, y: 5.0)
        if somePoint.isToTheRightOf(x: 1.0) {
            let something = 42
        }
    }
}
struct Point {
    var x = 0.0, y = 0.0
    func isToTheRightOf(x: Double) -> Bool {
        return self.x > x
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/StructSelfDisambiguity.cpp
 
#include <iostream>

struct Point {
    double x = 0.0, y = 0.0;
    bool isToTheRightOf(double x) {
        return (this->x > x);
    }
};

int main() {
    Point somePoint {4.0, 5.0};
    if (somePoint.isToTheRightOf(1.0)) 
        int something {42};

    return 0;
}